lib/collection/src/operations/verification/query.rs

```diff
@@ -35,6 +35,52 @@ impl Query {
         }
         Ok(())
     }
+
+    /// Check that the query does not perform a fullscan based on the collection configuration.
+    async fn check_fullscan(
+        &self,
+        using: &str,
+        collection: &Collection,
+        strict_mode_config: &StrictModeConfig,
+    ) -> CollectionResult<()> {
+        // Check only applies on `unindexed_filtering_retrieve`
+        if strict_mode_config.unindexed_filtering_retrieve == Some(false) {
+            if let Query::Vector(_) = &self {
+                let config = collection.collection_config.read().await;
```

Comment:
Maybe a bit too cautious but are you sure that this block call is not too slow for getting called in every request?
I don't know if there is a proper workaround though.

Reply:
In my experience the async lock on the collection configuration is not an issue.